pkg/controller/validation/policy/client.go

```diff
@@ -307,10 +315,10 @@ type Worker struct {
 // Process input queue. Validation delegated to the
 // policy agent.
 func (r *Worker) run() {
-	defer func() {
-		_ = recover()
-	}()
 	go func() {
+		defer func() {
+			_ = recover()
+		}()
```

Comment:
The `recover()` needs to be deferred in the function to prevent the panic if the channel is closed.